packages/react-components/src/components/Attachment/AttachmentCard.tsx

```diff
@@ -104,7 +102,9 @@ export const _AttachmentCard = (props: _AttachmentCardProps): JSX.Element => {
               iconName={
                 getFileTypeIconProps({
                   extension: useMemo((): string => {
```

Comment:
this `useMemo` looks a little bit strange and we should get warnings about it as hooks should be used at the top level of your React function, before any early returns
https://react.dev/reference/rules/rules-of-hooks


Could you please move it?